Here is a patch for testing.

  Ralf

Signed-off-by: Ralf Baechle <ralf@linux-mips.org>

 arch/mips/kernel/linux32.c     |    7 -------
 arch/mips/kernel/scall64-o32.S |    2 +-
 fs/read_write.c                |    4 ++--
 3 files changed, 3 insertions(+), 10 deletions(-)

Backport to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN a/arch/mips/kernel/linux32.c b/arch/mips/kernel/linux32.c
--- a/arch/mips/kernel/linux32.c	2009-03-24 14:21:04.000000000 -0600
+++ b/arch/mips/kernel/linux32.c	2009-03-24 14:23:58.000000000 -0600
@@ -253,13 +253,6 @@ sys32_settimeofday(struct compat_timeval
 	return do_sys_settimeofday(tv ? &kts : NULL, tz ? &ktz : NULL);
 }
 
-SYSCALL_DEFINE5(32_llseek, unsigned long, fd, unsigned long, offset_high,
-	unsigned long, offset_low, loff_t __user *, result,
-	unsigned long, origin)
-{
-	return sys_llseek(fd, offset_high, offset_low, result, origin);
-}
-
 /* From the Single Unix Spec: pread & pwrite act like lseek to pos + op +
    lseek back to original location.  They fail just like lseek does on
    non-seekable files.  */
diff -urpN a/arch/mips/kernel/scall64-o32.S b/arch/mips/kernel/scall64-o32.S
--- a/arch/mips/kernel/scall64-o32.S	2009-03-24 14:21:04.000000000 -0600
+++ b/arch/mips/kernel/scall64-o32.S	2009-03-24 14:23:58.000000000 -0600
@@ -343,7 +343,7 @@ sys_call_table:
 	PTR	sys_ni_syscall	 		/* for afs_syscall */
 	PTR	sys_setfsuid
 	PTR	sys_setfsgid
-	PTR	sys_32_llseek			/* 4140 */
+	PTR	sys_llseek			/* 4140 */
 	PTR	compat_sys_getdents
 	PTR	compat_sys_select
 	PTR	sys_flock
diff -urpN a/fs/read_write.c b/fs/read_write.c
--- a/fs/read_write.c	2009-03-24 14:21:04.000000000 -0600
+++ b/fs/read_write.c	2009-03-24 14:23:58.000000000 -0600
@@ -152,8 +152,8 @@ bad:
 }
 
 #ifdef __ARCH_WANT_SYS_LLSEEK
-SYSCALL_DEFINE5(llseek, unsigned int, fd, unsigned long, offset_high,
-		unsigned long, offset_low, loff_t __user *, result,
+SYSCALL_DEFINE5(llseek, unsigned int, fd, unsigned int, offset_high,
+		unsigned int, offset_low, loff_t __user *, result,
 		unsigned int, origin)
 {
 	int retval;
diff -urpN a/include/linux/syscalls.h b/include/linux/syscalls.h
--- a/include/linux/syscalls.h	2009-03-24 14:21:04.000000000 -0600
+++ b/include/linux/syscalls.h	2009-03-24 14:25:05.000000000 -0600
@@ -443,8 +443,8 @@ asmlinkage long sys_utimes(char __user *
 				struct timeval __user *utimes);
 asmlinkage long sys_lseek(unsigned int fd, off_t offset,
 			  unsigned int origin);
-asmlinkage long sys_llseek(unsigned int fd, unsigned long offset_high,
-			unsigned long offset_low, loff_t __user *result,
+asmlinkage long sys_llseek(unsigned int fd, unsigned int offset_high,
+			unsigned int offset_low, loff_t __user *result,
 			unsigned int origin);
 asmlinkage long sys_read(unsigned int fd, char __user *buf, size_t count);
 asmlinkage long sys_readahead(int fd, loff_t offset, size_t count);
